wled00/FX_fcn.cpp

```diff
@@ -152,6 +153,27 @@ Segment& Segment::operator= (Segment &&orig) noexcept {
   return *this;
 }
 
+// allocate and/or resize effect buffers if necessary
+void Segment::allocateBuffers() {
+  uint16_t len = virtualLength();
```

Comment:
This will yield unnecessary heap fragmentation. Please use `length()` instead as `length()` does not change when changes spacing, grouping or mirroring.